### ISSUE TITLE
`prefer-spread` is triggering on non-array `concat()` methods

### ISSUE THREAD

**Issue by tylerlaprade** [oxc-project/oxc#6034]
This is my RTK (Redux Toolkit) store. `prefer-spread` is complaining about concat, but this isn't an array, it's a custom util type called `Tuple` from [here](https://github.com/reduxjs/redux-toolkit/blob/8178e7f9cf4a0af08a34950aa9f6ea734fbd19b4/packages/toolkit/src/utils.ts#L42) that implements its own `concat()` method. There is no way to use a spread here.
```tsx
const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(baseApi.middleware, rtkQueryErrorLogger)
});
```

**Comment by DonIsaac** [comment]
Unfortunately, without type information, Oxlint is unable to tell `.concat()` is a custom, non-array method. Consider disabling it with `// oxlint-disable-next-lline prefer-spread`.

**Comment by tylerlaprade** [comment]
I see, thank you. Is there a plan to incorporate type information in the future?

**Comment by DonIsaac** [comment]
Not right now. It's an almost impossible task. https://github.com/swc-project/swc/issues/571

**Comment by tylerlaprade** [comment]
Interesting, I hadn't thought about that! I wonder if it makes sense to collaborate with Biome and/or QuickLint to develop a shared minimal type-parser for linting purposes...
https://github.com/biomejs/biome/issues/3187
